resources/js/features/forums/utils/preProcessShortcodesInBody.ts

```diff
@@ -30,7 +30,11 @@ const createPatterns = (type: string) => [
 ];
 
 export function preProcessShortcodesInBody(body: string): string {
-  let result = body;
+  // First, normalize any escaped newlines back to actual newlines.
+  let result = body.replace(/↵\n/g, '\n');
```

Comment:
A unicode character for escaping a newline? I've never seen that before.

Should you use an escape sequence for it (`\u21B5`) to prevent possible encoding issues?

Reply:
No harm in adding an escape sequence. Updated in latest - I'll go ahead and merge so the changes in this branch can safely land in #3074.